### PR TITLE
Improve documentation for AbstractNotification class

### PR DESCRIPTION
**Pull request by bess**
Trying to improve the documentation for the AbstractNotification class. I am troubleshooting how our notifications were working and having trouble understanding exactly what's going on here.

For example, I could not figure out what kind of object "comment" is expecting. @laritakr or @jeremyf? 

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by laritakr** [comment]
@bess I'm not positive, but I would expect it to be a Sipity::Comment https://github.com/samvera/hyrax/blob/master/app/models/sipity/comment.rb#L1-L13

File: app/services/hyrax/workflow/abstract_notification.rb [hunk 1]
```diff
@@ -1,5 +1,27 @@
 module Hyrax
   module Workflow
+    # Subclass AbstractNotification to create a notification.
```
Comment (jcoyne) [inline app/services/hyrax/workflow/abstract_notification.rb]: please specify that it's a notification that happens when a state transition occurs. You probably want to use the `@abstract` YARD tag too.

File: app/services/hyrax/workflow/abstract_notification.rb [hunk 2]
```diff
@@ -9,6 +31,10 @@ def self.send_notification(entity:, comment:, user:, recipients:)
 
       attr_reader :work_id, :title, :comment, :user, :recipients
 
+      # @param [Sipity::Entity] entity the Sipity::Entity that is a proxy for the relevant Hyrax work
+      # @param [?] comment
```
Comment (jcoyne) [inline app/services/hyrax/workflow/abstract_notification.rb]: This method is called here: https://github.com/samvera/hyrax/blob/520411c17903f3ab294fa7306d5877f1ccf82344/app/services/hyrax/workflow/notification_service.rb#L39 and in that class it specifies that comment is an object that responds to `#comment` https://github.com/samvera/hyrax/blob/520411c17903f3ab294fa7306d5877f1ccf82344/app/services/hyrax/workflow/notification_service.rb#L15

**Comment by bess** [comment]
@jcoyne and @laritakr : Changes incorporated. Thank you!

File: app/services/hyrax/workflow/abstract_notification.rb [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+    #
+    #           def subject
+    #             "Deposit #{title} has been approved"
+    #           end
+    #
+    #           def message
+    #             "#{title} (#{link_to work_id, document_path}) has been approved by #{user.display_name}  #{comment}"
+    #           end
+    #
+    #           # Add the user who initiatied this action to the list of users being notified
```
Comment (mjgiarlo) [inline app/services/hyrax/workflow/abstract_notification.rb]: Typo in comment: `s/initiatied/initiated/`

File: app/services/hyrax/workflow/abstract_notification.rb [hunk 4]
```diff
@@ -9,6 +31,10 @@ def self.send_notification(entity:, comment:, user:, recipients:)
 
       attr_reader :work_id, :title, :comment, :user, :recipients
 
+      # @param [Sipity::Entity] entity - the Sipity::Entity that is a proxy for the relevant Hyrax work
+      # @param [#comment] comment - the comment associated with the action being taken, could be a Sipity::Comment, or anything that responds to a #comment method
+      # @param [Hyrax::User] user - the user who has performed the relevant action
+      # @param [Hash] recipients - a hash with keys "to" and (optionally) "cc", each of which contains an Array of Users to notify
```
Comment (mjgiarlo) [inline app/services/hyrax/workflow/abstract_notification.rb]: You can also add the options to YARD, and abridge the comment about `recipients`, via:

```
# @option recipients [Array<Hyrax::User>] :to a list of users to which to send the notification
# @option recipients [Array<Hyrax::User>] :cc a list of users to which to copy on the notification
```

Comment (bess) [inline app/services/hyrax/workflow/abstract_notification.rb]: @mjgiarlo Suggestions incorporated, thanks!

**Comment by mjgiarlo** [comment]
@bess Thanks! Looks great. :white_check_mark:

**Pull request merged**